src/fsharp/NameResolution.fs

```diff
@@ -1498,15 +1498,47 @@ type TcSymbolUseData =
 /// are allocated (one per file), but they are large because the allUsesOfAllSymbols array is large.
 type TcSymbolUses(g, capturedNameResolutions : ResizeArray<CapturedNameResolution>, formatSpecifierLocations: (range * int)[]) = 
     
+    let chunks (items: ResizeArray<'t>) chunkSize =
```

Comment:
Two things

1. Move all the generic logic to a new helper in `illib.fs` called `ResizeArray.mapToSmallArrayChunks : ('T1 -> 'T2) -> IList<'T1> -> 'T2[][]`

2. Hence remove the introduction of the intermediate array `chunkedResolutions`, i.e. just

    let allUsesOfSymbols = chunkedResolutions |> ResizeArray.mapToSmallArrayChunks (fun cnr -> { Item=cnr.Item; ItemOccurence=cnr.ItemOccurence; DisplayEnv=cnr.DisplayEnv; Range=cnr.Range } )

I'm sure we will end up re-using this somewhere.

Reply:
Could we go further and introduce a data structure that does all it internally while implementing `IList<T>` so it's easy to use in client code?

Reply:
I was a bit curious about this, especially given that `TcResultsSinkImpl`'s `capturedMethodGroupResolutions` list grows at the same rate as `capturedNameResolutions` which is the source of the huge `ResizeArray` already.

Reply:
@auduchinok That could get a little tricky. Some data (e.g., a chunked string) may need to specialize on what "chunking" means for it. Not opposed to the idea though